tools/api_versioning/utils.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python
+"""Utility functions for accessing the API version (X.Y.Z format).
+"""
+from collections import namedtuple
+import pathlib
+import string
+import sys
+
+ApiVersion = namedtuple('ApiVersion', ['major', 'minor', 'patch'])
+
+
+def get_api_version(input_path):
+    """Returns the API version from a given API version input file.
+
+    Args:
+      input_path: the file containing the API version (API_VERSION).
+
+    Returns:
+      a namedtuple containing the major, minor, patch versions.
+    """
+    lines = pathlib.Path(input_path).read_text().splitlines()
+    assert len(lines) == 1
```

Comment:
also if this triggers it will raise an assertion errors about lengh which might not be that helpful

Reply:
I think an assertion/exception is needed in case the API_VERSION file contents are accidentally modified.
I've added a comment to the assertion failure, but let me know if you think it should be removed.

Reply:
if you keep it then add an exception that will be useful when the code runs

with just an `assert` you would get something like this

```python
>>> assert len([]) == 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

ie no useful info

i think you need to use an error that will provide info, eg:

```python
if len(lines) != 1:
    raise APIHeaderError("Explanation of what has gone wrong")
```

not sure exactly what kind of error or message is best

Reply:
Hmmm, interesting...
 I was under the impression that assertion failure also prints the line number. This is what I get at the moment:
 ```
 Traceback (most recent call last):
  File "/home/adip/git/envoy/bazel-bin/tools/protoxform/protoprint.runfiles/envoy/tools/protoxform/protoprint.py", line 733, in <module>
    traverse.traverse_file(file_proto, ProtoFormatVisitor(api_version_file_path, frozen_proto)))
  File "/home/adip/git/envoy/bazel-bin/tools/protoxform/protoprint.runfiles/envoy/tools/protoxform/protoprint.py", line 586, in __init__
    current_api_version = api_version_utils.get_api_version(api_version_file_path)
  File "/home/adip/git/envoy/bazel-bin/tools/protoxform/protoprint.runfiles/envoy/tools/api_versioning/utils.py", line 28, in get_api_version
    assert len(lines) == 1, 'Api version file must have a single line of format X.Y.Z, where X, Y, and Z are non-negative integers.'
AssertionError: Api version file must have a single line of format X.Y.Z, where X, Y, and Z are non-negative integers.
```

I don't object using an exception, although there are other errors that may emit an assertion failure (e.g., empty lines), so I'll just add explicit handling of these.

Reply:
yep - use a proper exception - its also catchable then